### DISCUSSION TITLE
Follow system theme

### DISCUSSION THREAD

**Discussion by Draculah** [stylus/stylus#2677] [Q&A] [answered]
Hello,

I think it would be useful if there would be an option that allows user to specify which theme to use and when. This way, users can specify themselves a dark theme, and when on lit places, change their desktop with some shortcut (which I'm doing using the [Gnome Night Switcher](https://extensions.gnome.org/extension/2236/night-theme-switcher/) when I am in lit places) to change all theming. [Dark Reader already has such an option](https://darkreader.org/blog/automation/), and it now comes in very handy now.

I propose that the user would be able to set 2 themes (light & dark) that show depending on the system theme, coming with the following extra functionality:

- Lock dark and light themes: changing either would change both, and thus implicitly, the system theme is ignored
- Have a 'null' / unset theme
  - if light theme for a site is not set and the system theme is light, then the site's theme will be followed
  - if dark theme for a site is not set and the system theme is dark, then the site's theme will be followed (which may have its own auto light/dark theme, or has simply 'dark' in it's cookie preferences)

It's of course desirable that the themes themselves support light and dark modes to have a consistent look & feel, but the ones I need rarely do, so having the lock on dark & light by default gives the user the opportunity to specify manually what they want in which situation.

**Comment by vendethiel** [comment, marked as answer]
Wrong stylus repository.
